crates/sparrow-compiler/src/dfg.rs

```diff
@@ -125,6 +125,14 @@ impl Dfg {
         self.add_expression(Expression::Inst(InstKind::Simple(instruction)), children)
     }
 
+    pub(super) fn add_udf(
```

Comment:
Do we need this? Should `ast_to_dfg` just call `add_expression` directly?

Reply:
We could, but I like keeping it. It abstracts away the need for the caller to know the types of the `Expression` and `InstKind` in order to create the UDF. Similar to above, in `add_instruction`:

```
    pub(super) fn add_instruction(
        &mut self,
        instruction: InstOp,
        children: ChildrenVec,
    ) -> anyhow::Result<Id> {
        self.add_expression(Expression::Inst(InstKind::Simple(instruction)), children)
    }
```